src/TRoomDB.cpp

```diff
@@ -255,10 +255,9 @@ bool TRoomDB::__removeRoom(int id)
             ++i;
         }
         rooms.remove(id);
-        if (roomIDToHash.contains(id)) {
-            QString hash = roomIDToHash[id];
-            roomIDToHash.remove(id);
-            hashToRoomID.remove(hash);
+        // Always remove any matching hashToRoomID entries
+        if (!pR->mHash.isEmpty()) {
+            hashToRoomID.remove(pR->mHash, id);
```

Comment:
> This will leak.

I am not sure I agree, the action of ***setting*** a hash for a roomId should always clear the previous hash for the same roomId that is stored in the `TRoom` (and at least development releases of Mudlet will moan if the hash is not cleared for that roomId).